llama_index/vector_stores/redis.py

```diff
@@ -0,0 +1,323 @@
+            )
+
+            mapping = {
+                "id": result.id,
+                "doc_id": result.ref_doc_id,
+                "text": result.node.text,
+                self._vector_key: array_to_buffer(result.embedding),
+                **node_info,
+                **extra_info,
+            }
+            ids.append(result.id)
+            key = "_".join((self._prefix, str(result.id)))
+            self._redis_client.hset(key, mapping=mapping)
+
+        _logger.info(f"Added {len(ids)} documents to index {self._index_name}")
+        return ids
+
+    def delete(self, doc_id: str) -> None:
+        """Delete a specific document from the index by doc_id
+
+        Args:
+            doc_id (str): The doc_id of the document to delete.
+        """
+        # use tokenizer to escape dashes in query
+        query_str = "@doc_id:{%s}" % self.tokenizer.escape(doc_id)
+        # find all documents that match a doc_id
+        results = self._redis_client.ft(self._index_name).search(query_str)
+
+        for doc in results.docs:
+            self._redis_client.delete(doc.id)
+        _logger.info(
+            f"Deleted {len(results.docs)} documents from index {self._index_name}"
+        )
+
+    def delete_index(self):
+        """Delete the index and all documents."""
+        _logger.info(f"Deleting index {self._index_name}")
+        self._redis_client.ft(self._index_name).dropindex(delete_documents=True)
+
+    def query(self, query: VectorStoreQuery) -> VectorStoreQueryResult:
+        """Query the index.
+
+        Args:
+            query (VectorStoreQuery): query object
+
+        Returns:
+            VectorStoreQueryResult: query result
+        """
+        from redis.exceptions import ResponseError as RedisResponseError
+
+        redis_query = self._prep_query(query.similarity_top_k)
+        query_params = {
+            "vector": array_to_buffer(query.query_embedding),
+        }
+        _logger.info(f"Querying index {self._index_name}")
+
+        try:
+            results = self._redis_client.ft(self._index_name).search(
+                redis_query, query_params=query_params
+            )
+        except RedisResponseError as e:
+            _logger.error(f"Error querying index {self._index_name}: {e}")
+            raise e
+
+        ids = []
+        nodes = []
+        scores = []
+        for doc in results.docs:
+            node = Node(
+                text=doc.text,
+                doc_id=doc.id,
+                embedding=None,
+                relationships={DocumentRelationship.SOURCE: doc.doc_id},
+            )
+            ids.append(doc.id)
+            nodes.append(node)
+            scores.append(1 - float(doc.vector_score))
+        _logger.info(f"Found {len(nodes)} results for query with id {ids}")
+
+        return VectorStoreQueryResult(nodes=nodes, ids=ids, similarities=scores)
+
+    def persist(self, in_background=True) -> None:
```

Comment:
right now `VectorStore` defines `persist` with a `persist_path` param:
```python 
    def persist(self, persist_path: str) -> None:
        return None
```
This is because we provide a convenience method to persist to disk via `StorageContext`, which tries to save everything into the same folder by configuring and passing the persist_path.

Could we just take in the arg and ignore it here?

Reply:
done

Reply:
wasn't exactly sure what you wanted/mypy wanted here, but I took a crack at it. Happy to make this a different method if that's what you meant.

Reply:
wasn't exactly sure what you wanted/mypy wanted here, but I took a crack at it. Happy to make this a different method if that's what you meant.